src/ts/components/subtitleoverlay.ts

```diff
@@ -507,6 +538,13 @@ export class SubtitleRegionContainerManager {
     this.subtitleRegionContainers[regionContainerId].addLabel(label, overlaySize);
   }
 
+  updateLabel(previousLabel: SubtitleLabel, newLabel: SubtitleLabel): void {
+    const { regionContainerId } = this.getRegion(previousLabel);
+
+    this.subtitleRegionContainers[regionContainerId].removeLabel(previousLabel);
```

Comment:
Question: here we implement a different logic than the FIFO approach above, is this correct?

Reply:
Unless I am missing something the FIFO approach should still be valid here.